app/app_utils.c

```diff
@@ -201,10 +203,12 @@ static ACVP_RESULT totp(char **token, int token_max) {
 
     otp = bin % DIGITS_POWER[ACVP_TOTP_LENGTH];
 
-    // generate format string like "%06d" to fix digits using 0
-    sprintf(format, "%c0%ldd", '%', (long int)ACVP_TOTP_LENGTH);
-
-    sprintf((char *)token_buff, format, otp);
+    /* If we ever change the size then it will fail hard here */
+    if (ACVP_TOTP_LENGTH != 8) {
```

Comment:
Is this something we may want to make a #define for?

Reply:
Will address in next PR, thanks